pyi.py

```diff
@@ -159,7 +159,40 @@ class TypeVarInfo(NamedTuple):
 )
 
 
+class PyflakesPreProcessor(ast.NodeTransformer):
+    """Transform AST prior to passing it to pyflakes.
+
+    This reduces false positives on recursive class definitions.
+    """
+
+    def visit_ClassDef(self, node: ast.ClassDef) -> ast.ClassDef:
+        self.generic_visit(node)
+        node.bases = [
+            # Remove the subscript to prevent F821 errors from being raised for (valid) recursive definitions.
+            base.value if isinstance(base, ast.Subscript) else base
+            for base in node.bases
+        ]
+        return node
+
+
 class PyiAwareFlakesChecker(FlakesChecker):
+    def __init__(self, tree: ast.AST, **kwargs: Any) -> None:
+        super().__init__(PyflakesPreProcessor().visit(tree), **kwargs)
+
+    @property
+    def annotationsFutureEnabled(self):
+        """pyflakes can already handle forward refs for annotations, but only via
+        `from __future__ import annotations`.
+
+        We don't want to bother including this in every file, so we just set this to `True`.
+        """
+        return True
+
+    @annotationsFutureEnabled.setter
+    def annotationsFutureEnabled(self, value: bool):
+        """noop, we never want to change the value to anything but `True` anyway."""
```

Comment:
Pet peeve: I hate the word "noop" -- it feels really tech-jargony to me :)

```suggestion
        """Does nothing, as we always want this property to be `True`."""
```